src/views/studio/CreateEditChannelView/CreateEditChannelView.tsx

```diff
@@ -46,220 +53,240 @@ type CreateEditChannelViewProps = {
 }
 
 const CreateEditChannelView: React.FC<CreateEditChannelViewProps> = ({ newChannel }) => {
-  // TODO Add hook for fetching currently active channel
-  const id = '1494'
-  const { channel, loading, error } = useChannel(id, { skip: newChannel })
+  const {
+    activeUser: { channelId },
+  } = useActiveUser()
+
+  const { channel, loading, error } = useChannel(channelId || '', { skip: newChannel || !channelId })
 
   const {
     register,
-    handleSubmit: useHandleSubmit,
+    handleSubmit: createSubmitHandler,
     control,
-    setValue,
-    getValues,
-    formState: { isDirty },
+    formState: { isDirty, dirtyFields },
     reset,
     errors,
   } = useForm<Inputs>({
     defaultValues: {
       avatar: { url: null, blob: null },
       cover: { url: null, blob: null },
-      channelName: '',
+      title: '',
       description: '',
-      selectedLanguage: languages[0],
+      language: languages[0].value,
       isPublic: true,
     },
   })
 
   const avatarDialogRef = useRef<ImageCropDialogImperativeHandle>(null)
   const coverDialogRef = useRef<ImageCropDialogImperativeHandle>(null)
 
-  const [avatarImageUrl, setAvatarImageUrl] = useState<string | null>(null)
-  const [coverImageUrl, setCoverImageUrl] = useState<string | null>(null)
-  const [isActionBarActive, setActionBarActive] = useState(false)
+  const [transactionStatus, setTransactionStatus] = useState<ExtrinsicStatus | null>(null)
 
   useEffect(() => {
     if (loading || newChannel || !channel) {
       return
     }
-    const { avatarPhotoUrl, coverPhotoUrl, title, description, isPublic, language } = channel
-    const foundLanguage = languages.find(({ name }) => name === language?.name)
+
+    const {
+      avatarPhotoUrl,
+      avatarPhotoAvailability,
+      avatarPhotoDataObject,
+      coverPhotoUrl,
+      coverPhotoAvailability,
+      coverPhotoDataObject,
+      title,
+      description,
+      isPublic,
+      language,
+    } = channel
+
+    const avatarAssetUrl = createUrlFromAsset(avatarPhotoAvailability, avatarPhotoUrl, avatarPhotoDataObject)
+    const coverAssetUrl = createUrlFromAsset(coverPhotoAvailability, coverPhotoUrl, coverPhotoDataObject)
+
+    const foundLanguage = languages.find(({ value }) => value === language?.iso)
+
     reset({
-      ...getValues(),
```

Comment:
@mikkio-j Was this needed for some specific case? It seems to work fine without it

Reply:
@kdembler shouldn't be needed anymore, I added that before `description`, `isPublic`, `selectedLanguage` fields were introduced (its purpose was to include all fields in reset - if the field was not provided then it would get that from `defaultValues`)

Reply:
I think it should work the same way, since missing values should get populated with defaults anyway. Let's see then, if it breaks something we'll reintroduce it